File: python/test/quirks_test.py
from aslan.aslan_parser import ASLANParser

class TestASLANParserQuirks:
    def setup_method(self):
        self.parser = ASLANParser()

    def test_parses_empty_string(self):
        result = self.parser.parse('')
        assert result == {
            '_default': '',
        }

    def test_parses_string_starting_with_delimiter(self):
        result = self.parser.parse('[asland_test]test')
        assert result == {
            '_default': None,
            'test': 'test',
        }

File: rust/tests/quirks_test.rs
use aslan::ASLANParser;
use serde_json::json;

#[test]
fn parses_empty_string() {
    let mut parser = ASLANParser::new();
    let result = parser.parse("");
    assert_eq!(result, json!({
        "_default": ""
    }));
}

#[test]
fn parses_string_starting_with_delimiter() {
    let mut parser = ASLANParser::new();
    let result = parser.parse("[asland_test]test");
    assert_eq!(result, json!({
        "_default": null,
        "test": "test"
    }));
}
